lib/active_merchant/billing/gateways/cashnet.rb

```diff
@@ -117,11 +117,11 @@ def add_invoice(post, money, options)
           codes_and_amounts = options[:item_codes].transform_keys { |key| key.to_s.delete('_') }
           codes_and_amounts.each do |key, value|
             post[key] = value if key.start_with?('itemcode')
-            post[key] = amount(value) if key.start_with?('amount')
+            post[key] = amount(value.to_i) if key.start_with?('amount')
           end
         else
           post[:itemcode] = (options[:item_code] || @options[:default_item_code])
-          post[:amount] = amount(money)
+          post[:amount] = amount(money.to_i)
```

Comment:
Same as above: The amount method seems to be then transforming this back into a string 🤦🏻  This was discovered by adding a puts statement to confirm the class. I think instead that the fix would be `amount(money).to_i`

Reply:
so the Cashnet gateway actually WANTS this in a string, with cents included such as this: `1001` converts to `'10.01'`. The reason we made this conversion is because when we were passing a hash with nested `amount` and `item_code`s in it, the amounts were getting converted to strings, which WOULD be fine if we also didn't need to transform their amounts into what Cashnet expects.

We can discuss in pairing, but basically slapping the `to_i` method onto the money value is ensuring that IF money is passed as a string (which it shouldn't be), than we convert it back into an integer before running the `amount` method on it.

Reply:
Ah, thank you for the explanation! This is great. There are 2 failing remote tests then, but they are also happening on the `master` branch so you're good to go!

Reply:
yeah, those darn remote tests :/ we noticed those two, but decided to kick that can down the road 😂 thanks for your feedback!!